app/src/main/java/campus/tech/kakao/map/viewmodel/PlaceViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package campus.tech.kakao.map.viewmodel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import campus.tech.kakao.map.data.PlaceRepository
+import campus.tech.kakao.map.model.Place
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class PlaceViewModel(application: Application) : AndroidViewModel(application) {
+    private val placeRepository: PlaceRepository = PlaceRepository(application)
+    private val _searchResults = MutableLiveData<List<Place>>()
+
+    val searchResults: LiveData<List<Place>> get() = _searchResults
+
+    init {
+        clearSearchResults()
+    }
+
+    fun insertPlace(place: Place) {
+        placeRepository.insertPlace(place)
+    }
+
+    fun clearAllPlaces() {
+        placeRepository.clearAllPlaces()
+    }
+
+    private fun clearSearchResults() {
+        _searchResults.value = emptyList()
+    }
+
+    fun searchPlacesByCategory(category: String) {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                val results = placeRepository.getPlacesByCategory(category)
+                _searchResults.postValue(results)
+            }
+        }
+    }
```

Comment:
Dispatchers.IO 처리는 잘해주셨어요. 다만 withContext 로 처리하기 보다는 launch 시점에 context 를 설정하는 것이 어떨까요?

```suggestion
    fun searchPlacesByCategory(category: String) {
        viewModelScope.launch(Dispatchers.IO) {
            val results = placeRepository.getPlacesByCategory(category)
            _searchResults.postValue(results)
        }
    }
```

Reply:
더불어 viewModelScope 은 launch 시에 아무것도 설정하지 않으면 어떤 쓰레드에서 동작되는지 확인해보시면 좋을 것 같아요!